extractors/cds/tools/src/cds/compiler/command.ts

```diff
@@ -1,12 +1,93 @@
 import { execFileSync } from 'child_process';
 import { existsSync, readdirSync } from 'fs';
-import { join } from 'path';
-
-import { quote } from 'shell-quote';
+import { join, resolve } from 'path';
 
+import type { ValidatedCdsCommand } from './types';
 import { fileExists } from '../../filesystem';
 import { cdsExtractorLog } from '../../logging';
 
+/**
+ * Predefined secure CDS command patterns
+ */
+const ALLOWED_CDS_COMMANDS = {
```

Comment:
I don't believe we actually need to implement an allow list for cds commands. Reviewing the code, I believe the command tested against this list is exclusively generated by the `getBestCdsCommand`. The command provided by that function is either already one of the items from this list, or it is a command created from a fixed path prefixed with the cache directory, in which case it is not checked against this list anyway.

Instead of an allow list, I think we should just generate the "validated commands" directly - e.g. an object with the `executable` and `args` properties.